examples/gno.land/r/gov/dao/v3/impl/prop_requests.gno

```diff
@@ -0,0 +1,208 @@
+	_, ok := memberstore.Tiers.GetTier(tier)
+	if !ok {
+		panic("provided tier does not exists")
+	}
+
+	if tier != memberstore.T1 && tier != memberstore.T2 {
+		panic("Only T1 and T2 members can be added by proposal. To add a T3 member use AddMember function directly.")
+	}
+
+	if portfolio == "" {
+		panic("A portfolio for the proposed member is required")
+	}
+
+	member, _ := memberstore.Get().GetMember(std.OriginCaller())
+	if member == nil {
+		panic("proposer is not a member")
+	}
+
+	if member.InvitationPoints <= 0 {
+		panic("proposer does not have enough invitation points for inviting new people to the board")
+	}
+
+	return dao.NewProposalRequestBuilder("New Member Proposal").
+		WithDescription("This proposal is looking to add a new member to the board.").
+		WithCallback(addMemberCallback).
+		WithMetadata(&AddMemberMetadata{
+			Member:    memberByTier(tier),
+			Portfolio: portfolio,
+			Tier:      tier,
+			Addr:      addr,
+			Caller:    member,
+		}).Build()
+}
+
+type RemoveMemberMetadata struct {
+	Addr   std.Address
+	Tier   string
+	Reason string
+}
+
+func (a *RemoveMemberMetadata) IsMetadata() {}
+func (a *RemoveMemberMetadata) String() string {
+	return ufmt.Sprintf("Member with address %v will be withdrawn.\n\n REASON: %v.", a.Addr, a.Reason)
+}
+
+func removeMemberCallback(m dao.Metadata) error {
+	rmm, ok := m.(*RemoveMemberMetadata)
+	if !ok {
+		panic("metadata type for removing a member is wrong")
+	}
+
+	memberstore.Get().RemoveMember(rmm.Addr)
+
+	return nil
+}
+
+func NewWithdrawMemberRequest(addr std.Address, reason string) dao.ProposalRequest {
+	member, tier := memberstore.Get().GetMember(addr)
+	if member == nil {
+		panic("user we want to remove not found")
+	}
+
+	if tier == memberstore.T1 && reason == "" {
+		panic("T1 user removals must contains a reason.")
+	}
+
+	return dao.NewProposalRequestBuilder("Member Withdrawn").
+		WithDescription("This proposal is looking to remove a member from the board.").
+		WithCallback(removeMemberCallback).
+		WithMetadata(&RemoveMemberMetadata{
+			Addr:   addr,
+			Tier:   tier,
+			Reason: reason,
+		}).Build()
+}
+
+type PromoteMemberMetadata struct {
+	FromTier string
+	ToTier   string
+	Addr     std.Address
+	Member   *memberstore.Member
+}
+
+func (a *PromoteMemberMetadata) IsMetadata() {}
+func (a *PromoteMemberMetadata) String() string {
+	return ufmt.Sprintf("A new member with address %v will be promoted from tier %v to tier %v.", a.Addr, a.FromTier, a.ToTier)
+}
+
+func promoteMemberCallback(m dao.Metadata) error {
+	pmm, ok := m.(*PromoteMemberMetadata)
+	if !ok {
+		panic("metadata type for promoting a member is wrong")
+	}
+
+	prevTier := memberstore.Get().RemoveMember(pmm.Addr)
+	if prevTier == "" {
+		panic("member not found, so cannot be promoted")
+	}
+
+	if prevTier != pmm.FromTier {
+		panic("previous tier changed from the one indicated in the proposal")
+	}
+
+	return memberstore.Get().SetMember(pmm.ToTier, pmm.Addr, pmm.Member)
```

Comment:
I believe we should panic here. This function could create strange behavior if everything works except for this last line, such as firing someone instead of promoting them.